src/analyses/memLeak.ml

```diff
@@ -20,6 +20,105 @@ struct
   let context _ d = d
 
   (* HELPER FUNCTIONS *)
+  let get_global_vars () =
+    (* Filtering by GVar seems to account for declarations, as well as definitions of global vars *)
+    List.filter_map (function GVar (v, _, _) -> Some v | _ -> None) !Cilfacade.current_file.globals
```

Comment:
Does CIL really guarantee that? I think in other places we don't assume it does and also handle `GVarDecl`.

Reply:
I did some manual inspection to check for that. Iirc, `GVarDecl` was giving me a bunch of stuff from the header files as well which seemed a bit too much for me. Anyhow, I can add `GVarDecl` in there as well if we don't have a CIL guarantee regarding `GVar`.

Reply:
Should be fixed in https://github.com/goblint/analyzer/pull/1241/commits/6cc01b5177b6bc31899c290b6ae65660bf4aa805